flytekit/tools/fast_registration.py

```diff
@@ -139,9 +141,22 @@ def fast_package(
                         filter=lambda x: tar_strip_file_attributes(x),
                     )
 
-            with gzip.GzipFile(filename=archive_fname, mode="wb", mtime=0) as gzipped:
-                with open(tar_path, "rb") as tar_file:
-                    gzipped.write(tar_file.read())
+            if shutil.which("pigz"):
+                with open(archive_fname, "wb") as gzipped:
+                    subprocess.run(["pigz", "-c", tar_path], stdout=gzipped, check=True)
```

Comment:
Can we store the output of `shutil.which` and use it here?

Reply:
https://github.com/flyteorg/flytekit/pull/2729/commits/d32822314c401205ee93b7801dcbd53af9301775